### DISCUSSION TITLE
Earth Rotation Matrix

### DISCUSSION THREAD

**Discussion by bnvm72279** [skyfielders/python-skyfield#659] [Q&A]
I am using skyfield to create an interactive model of the solar system and I would like to get the rotation of the planets, particularly the earth, added in. Is there a way to access this data? Thanks.

**Comment by brandon-rhodes** [comment]
Earth rotation is accurately modeled by Skyfield, and can be accessed through the `lst_hours_at()` method:

https://rhodesmill.org/skyfield/api-topos.html#skyfield.toposlib.GeographicPosition.lst_hours_at

Moon rotation is modeled; some routines that depend on it are here:

https://rhodesmill.org/skyfield/planetary.html

I haven’t tried the code against any other Solar System bodies yet. I can see that the data file used on the previous page has "PM" values for several other planets, which I assume give the angle of the prime meridian on those planets, and thus their rotation:

https://naif.jpl.nasa.gov/pub/naif/JUNO/kernels/pck/pck00008.tpc

But I've not had time to sit down and see how many of the planet rotations Skyfield can currently understand.
